apps/email/js/html_cache_restore.js

```diff
@@ -21,23 +66,143 @@ var startupCacheEventsSent = false;
-// true to the first call made to it.
-window.htmlCacheRestorePendingMessage = [];
+var startupOnModelLoaded;
 
 (function() {
-  var selfNode = document.querySelector('[data-loadsrc]'),
+  var evt,
+      handlers = {},
+      pendingMessage = null,
+      startingUp = true,
+      lastEntryTime = 0,
+      handlerQueues = {
+        notification: [],
+        activity: [],
+        'request-sync': []
+      },
+      selfNode = document.querySelector('[data-loadsrc]'),
       loader = selfNode.dataset.loader,
-      loadSrc = selfNode.dataset.loadsrc;
+      loadSrc = selfNode.dataset.loadsrc,
+      // Fragment ID parsing commented out for now until further testing can be
+      // done on non-trivial entry points, like manual_config.
+      startupData = {}; //parseFragmentId();
+
+  /**
+   * startupOnModelLoaded is set to this function if the
+   * data_has_account localStorage value is not known. Called eventually by
+   * config.js once the model module has been loaded.
+   */
+  function localOnModelLoaded(model, callback) {
+    model.once('acctsSlice', function(acctsSlice) {
+      // At this point, model will have set up 'data_has_account', so the the
+      // final view can be set and the rest of the world can start turning.
+      // Note that request-sync can get kicked off before this is all done,
+      // since the worker will have started up and registered to get those
+      // messages. It is OK though since not showing any UI. If this case is
+      // triggered, it will result in a UI display in that case, but that is OK,
+      // it only happens on an app update situation, from cookies to
+      // localStorage or some kind of localStorage reset. So a rare event, and
+      // does not cause harm, just a bit of extra work in those cases once.
+      console.log('localOnModelLoaded called, hasAccount: ' +
+                  localStorage.getItem('data_has_account'));
+
+      setDefaultView();
+      hydrateHtml(startupData.view);
+      startupOnModelLoaded = null;
+      callback();
+    });
+  }
+
+  if (!localStorage.getItem('data_has_account')) {
+    console.log('data_has_account unknown, asking for model load first');
+    startupOnModelLoaded = localOnModelLoaded;
+  }
+
+  function hasAccount() {
+    // var _1 = performance.now();
+    var has = localStorage.getItem('data_has_account') === 'yes';
+    // console.log('@@@ LOCALSTORAGE GET data_has_account: ' +
+    //             (performance.now() - _1));
+    return has;
+  }
+
+  function setDefaultView() {
+    if (hasAccount()) {
+      if (!startupData.view) {
+        startupData.view = 'message_list';
+      }
+    } else {
+      startupData.view = 'setup_account_info';
+    }
+  }
+
+  // Set up the default view, but if that is not possible to know yet, since
+  // the status of hasAccount is unknown, wait for the callback to set it up.
+  if (!startupOnModelLoaded) {
+    setDefaultView();
+  }
+
+  startupData.entry = 'default';
+
+  /*
```

Comment:
I just removed the code. It was not that complicated, so not worth taking up space. And not sure yet how the fragment ID work will shake out yet. I did file [bug 1137510](https://bugzilla.mozilla.org/show_bug.cgi?id=1137510) to track the fragment ID work. Removed in commit 7d8f41f19303f833552bc490fe6bf87e1349526a.